### PR TITLE
fixed incorrect decoding of header seal_fields. added tests. #1090

### PR DESCRIPTION
**Pull request by debris**

### CODE REVIEW

**Comment by gavofyork** [comment]
what makes you think this is incorrect?

unless something (everything) changed since i wrote this code, it is correct.

**Comment by debris** [comment]
please see this comment https://github.com/ethcore/parity/issues/1090#issuecomment-219620535 and tests attached to pr

File: ethcore/src/header.rs [hunk 1]
```diff
@@ -219,8 +219,8 @@ impl Header {
 		s.append(&self.timestamp);
 		s.append(&self.extra_data);
 		if let Seal::With = with_seal {
-			for b in &self.seal { 
-				s.append_raw(&b, 1); 
+			for b in &self.seal {
```
Comment (gavofyork) [inline ethcore/src/header.rs]: `self.seal` is a vector of post-RLP-encoded fields. It is *not* pre-RLP-encoded data. The reason for this is to allow for complex fields as part of the seal without having meta-RLP within the header.

Comment (debris) [inline ethcore/src/header.rs]: I see... than we have inconsistency in `HeaderView` struct, cause it returns pre-RLP-encoded data. I will update the pr.

**Comment by debris** [comment]
updated the pr

**Pull request merged**